lightning/src/sync.rs

```diff
@@ -50,18 +49,69 @@ impl Condvar {
 		Condvar { }
 	}
 
-	pub fn wait<'a, T>(&self, guard: MutexGuard<'a, T>) -> LockResult<MutexGuard<'a, T>> {
+	pub fn wait<'a, T>(&'a self, guard: MutexGuard<'a, T>) -> LockResult<MutexGuard<'a, T>> {
 		Ok(guard)
 	}
 
 	#[allow(unused)]
-	pub fn wait_timeout<'a, T>(
-		&self,
-		guard: MutexGuard<'a, T>,
-		_dur: Duration,
-	) -> LockResult<(MutexGuard<'a, T>, ())> {
+	pub fn wait_timeout<'a, T>(&'a self, guard: MutexGuard<'a, T>, _dur: Duration) -> LockResult<(MutexGuard<'a, T>, ())> {
 		Ok((guard, ()))
 	}
 
 	pub fn notify_all(&self) {}
-}
\ No newline at end of file
+}
+
+/// This no_std version of RwLock actually only allows one lock at a time, because
+/// we are not expecting reentrancy in our straight-line code.
```

Comment:
Is this comment about reentrancy still accurate given our Slack discussion on `ChannelManager::process_pending_events`? Maybe should say "one write lock"?

Reply:
yeah, obsolete